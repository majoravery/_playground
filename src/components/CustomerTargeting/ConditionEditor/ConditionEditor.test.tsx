// import React from 'react';
// import '@testing-library/jest-dom';
// import { render } from '@testing-library/react';
// import userEvent from '@testing-library/user-event';
// import { ConditionEditor } from './ConditionEditor';
// import { ConditionEditorProps } from '../types';
// import {
//   basicRulesMock,
//   defaultConditionMock,
//   conditionsMock,
//   ruleOptionsMock,
//   addedConditionMock,
//   freeTextConditionMock,
// } from '../__mocks__/mocks';

// export const defaultConditionEditorProps: ConditionEditorProps = {
//   section: basicRulesMock,
//   conditionKey: 'condition-key-0',
//   conditionIndex: 0,
//   ruleKey: 'rule-key-0',
//   options: ruleOptionsMock,
//   conditions: conditionsMock,
//   addNewMode: false,
//   setOptions: jest.fn(),
//   setConditions: jest.fn(),
//   setAddNewMode: jest.fn(),
// };

// describe('ConditionEditor', () => {
//   it('renders', () => {
//     const { asFragment } = render(<ConditionEditor {...defaultConditionEditorProps} />);
//     expect(asFragment()).toMatchSnapshot();
//   });

//   it('upon applying, triggers setCondition and setAddNewMode', () => {
//     const setConditionsMock = jest.fn();
//     const { getByTestId } = render(
//       <ConditionEditor {...defaultConditionEditorProps} setConditions={setConditionsMock} />
//     );
//     userEvent.click(getByTestId('tab-content-footer-select-btn'));
//     expect(setConditionsMock).toHaveBeenCalledTimes(1);
//   });

//   it('upon applying, triggers setAddNewMode if its adding a new condition', () => {
//     const setAddNewModeMock = jest.fn();
//     const { getByTestId } = render(
//       <ConditionEditor
//         {...defaultConditionEditorProps}
//         conditions={[]}
//         addNewMode={true}
//         setAddNewMode={setAddNewModeMock}
//       />
//     );
//     userEvent.click(getByTestId('tab-content-footer-select-btn'));
//     expect(setAddNewModeMock).toHaveBeenCalledTimes(1);
//   });

//   it('upon cancel, triggers setAddNewMode if its a new condition', () => {
//     const setConditionsMock = jest.fn();
//     const setAddNewModeMock = jest.fn();
//     const { getByTestId } = render(
//       <ConditionEditor
//         {...defaultConditionEditorProps}
//         conditions={[]}
//         addNewMode={true}
//         setConditions={setConditionsMock}
//         setAddNewMode={setAddNewModeMock}
//       />
//     );
//     userEvent.click(getByTestId('tab-content-footer-remove-btn'));
//     expect(setAddNewModeMock).toHaveBeenCalledTimes(1);
//     expect(setConditionsMock).toHaveBeenCalledTimes(0);
//   });

//   it('upon cancel, triggers setCondition if its an existing condition', () => {
//     const setConditionsMock = jest.fn();
//     const setAddNewModeMock = jest.fn();
//     const { getByTestId } = render(
//       <ConditionEditor
//         {...defaultConditionEditorProps}
//         setConditions={setConditionsMock}
//         setAddNewMods={setAddNewModeMock}
//       />
//     );
//     userEvent.click(getByTestId('tab-content-footer-remove-btn'));
//     expect(setAddNewModeMock).toHaveBeenCalledTimes(0);
//     expect(setConditionsMock).toHaveBeenCalledTimes(1);
//   });

//   it('upon delete, triggers setCondition', () => {
//     const setConditionsMock = jest.fn();
//     const { getByTestId } = render(
//       <ConditionEditor {...defaultConditionEditorProps} setConditions={setConditionsMock} />
//     );
//     userEvent.click(getByTestId('profile-selector-delete-button'));
//     expect(setConditionsMock).toHaveBeenCalledTimes(1);
//   });

//   it('updates upon changing field key', () => {
//     const { getByTestId } = render(<ConditionEditor {...defaultConditionEditorProps} />);
//     const fieldKeyInput = getByTestId('profile-editor-field-key-control');
//     const defaultSelection = basicRulesMock.rules[0].key;
//     expect(fieldKeyInput).toHaveValue(defaultSelection);

//     const SECOND_INDEX = 1;
//     const newSelection = basicRulesMock.rules[SECOND_INDEX].key;
//     userEvent.selectOptions(fieldKeyInput, newSelection);
//     expect(fieldKeyInput).toHaveValue(newSelection);
//   });

//   it('updates upon changing single value', () => {
//     const { getByTestId } = render(<ConditionEditor {...defaultConditionEditorProps} />);
//     const selectInput = getByTestId('profile-editor-select-control');
//     const NEW_SELECTION_MOCK = 'ios';
//     expect(selectInput).not.toHaveValue(NEW_SELECTION_MOCK);

//     userEvent.selectOptions(selectInput, NEW_SELECTION_MOCK);
//     expect(selectInput).toHaveValue(NEW_SELECTION_MOCK);
//   });

//   it('updates upon changing comparator for multiple values', () => {
//     const { getByTestId } = render(<ConditionEditor {...defaultConditionEditorProps} />);

//     const FIELD_KEY_WITH_MULTIPLE_VALUES_SUPPORT = 'profile.VisitAttributes.FirstPlatformDevice';
//     const fieldKeyInput = getByTestId('profile-editor-field-key-control');
//     userEvent.selectOptions(fieldKeyInput, FIELD_KEY_WITH_MULTIPLE_VALUES_SUPPORT);

//     const comparatorInput = getByTestId('profile-editor-comparator-control');
//     const MULTIPLE_VALUES_COMPARATOR = 'in';

//     userEvent.selectOptions(comparatorInput, MULTIPLE_VALUES_COMPARATOR);
//     const valueInput = getByTestId('profile-editor-multiple-select-control');
//     expect(fieldKeyInput).toHaveValue(FIELD_KEY_WITH_MULTIPLE_VALUES_SUPPORT);
//     expect(comparatorInput).toHaveValue(MULTIPLE_VALUES_COMPARATOR);
//     expect(valueInput).toHaveValue('0 selected');
//   });
// });

// describe('ConditionEditor: applying condition', () => {
//   afterEach(() => {
//     jest.clearAllMocks();
//   });

//   it('fire setCondition and addNewMode after confirming the prompt alert', () => {
//     jest.spyOn(window, 'confirm').mockImplementation(() => true);
//     const setConditionsMock = jest.fn();
//     const setAddNewModeMock = jest.fn();
//     const { getByTestId } = render(
//       <ConditionEditor
//         {...defaultConditionEditorProps}
//         conditions={[defaultConditionMock]}
//         addNewMode={true}
//         setConditions={setConditionsMock}
//         setAddNewMode={setAddNewModeMock}
//       />
//     );
//     userEvent.click(getByTestId('tab-content-footer-select-btn'));
//     const SINGLE_CALL_COUNT = 1;
//     expect(setConditionsMock).toHaveBeenCalledTimes(SINGLE_CALL_COUNT);
//     expect(setConditionsMock).toHaveBeenCalledWith(addedConditionMock);
//     expect(setAddNewModeMock).toHaveBeenCalledTimes(SINGLE_CALL_COUNT);
//     expect(setAddNewModeMock).toHaveBeenCalledWith(false);
//   });

//   it('should trim whitespaces from string values', () => {
//     const setConditionsMock = jest.fn();
//     const { getByTestId } = render(
//       <ConditionEditor
//         {...defaultConditionEditorProps}
//         conditions={[defaultConditionMock]}
//         addNewMode={true}
//         setConditions={setConditionsMock}
//         setAddNewMode={jest.fn()}
//       />
//     );
//     userEvent.selectOptions(getByTestId('profile-editor-field-key-control'), 'Free Text Input');
//     userEvent.type(getByTestId('profile-editor-text-input-control'), ' android ');
//     userEvent.click(getByTestId('tab-content-footer-select-btn'));
//     expect(setConditionsMock).toHaveBeenCalledTimes(1);
//     expect(setConditionsMock).toHaveBeenCalledWith(freeTextConditionMock);
//   });
// });
export {};
