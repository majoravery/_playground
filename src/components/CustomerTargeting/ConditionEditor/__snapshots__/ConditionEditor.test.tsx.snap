// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConditionEditor renders 1`] = `
<DocumentFragment>
  <p>
    Please select the conditions for customer order.
  </p>
  <div
    class="profile-selector"
    data-testid="profile-selector"
  >
    <button
      class="button--component ghost-grey-600 button--component--icon-only profile-selector-delete-button btn btn-light btn-sm"
      data-testid="profile-selector-delete-button"
      type="button"
    >
      <svg
        fill="none"
        height="14"
        opacity="1"
        stroke="#051022"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="3 6 5 6 21 6"
        />
        <path
          d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
        />
        <line
          x1="10"
          x2="10"
          y1="11"
          y2="17"
        />
        <line
          x1="14"
          x2="14"
          y1="11"
          y2="17"
        />
      </svg>
    </button>
    <div
      class="profile-selector-labels"
    >
      <label
        class="field-input form-label"
      >
        Select field
      </label>
      <label
        class="comparator-input form-label"
      >
        Comparator
      </label>
      <label
        class="value-input form-label"
      >
        Value
      </label>
    </div>
    <form
      class="was-validated"
      novalidate=""
    >
      <div
        class="profile-selector-row"
      >
        <div
          class="profile-selector-form-control field-input form-group"
        >
          <select
            class="form-control"
            data-testid="profile-editor-field-key-control"
            id="field-input"
            name="fieldKey"
          >
            <option
              value="profile.VisitAttributes.FirstPlatformDevice"
            >
              First Platform Device
            </option>
            <option
              value="profile.VisitAttributes.TotalVisits"
            >
              Total Visits
            </option>
            <option
              value="profile.VisitAttributes.LastVisitTimestamp"
            >
              Last Visit Timestamp
            </option>
            <option
              value="profile.VisitAttributes.IsGuest"
            >
              Guest
            </option>
            <option
              value="profile.OrderAttributes.NightHourOrderRate"
            >
              Night Hour Order Rate
            </option>
            <option
              value="dummyProfile.freeTextInput"
            >
              Free Text Input
            </option>
            <option
              value="profile.VisitAttributes.SignedUpDateLocal"
            >
              Signed Up Date
            </option>
          </select>
        </div>
        <div
          class="profile-selector-form-control comparator-input form-group"
        >
          <select
            class="form-control"
            data-testid="profile-editor-comparator-control"
            id="comparator-input"
            name="comparator"
          >
            <option
              value="="
            >
              =
            </option>
            <option
              value="in"
            >
              in
            </option>
          </select>
        </div>
        <div
          class="profile-selector-form-control value-input form-group"
        >
          <select
            class="form-control"
            data-testid="profile-editor-select-control"
            id="value-input"
            name="value"
          >
            <option
              value="android"
            >
              Android
            </option>
            <option
              value="ios"
            >
              iOS
            </option>
            <option
              value="mweb"
            >
              Mobile Web
            </option>
          </select>
        </div>
      </div>
    </form>
  </div>
  <div
    class="tab-content-footer"
  >
    <button
      class="button--component tab-content-footer-remove-btn btn btn-outline-primary"
      data-testid="tab-content-footer-remove-btn"
      type="button"
    >
      Cancel
    </button>
    <button
      class="button--component tab-content-footer-select-btn btn btn-primary"
      data-testid="tab-content-footer-select-btn"
      type="button"
    >
      Apply
    </button>
  </div>
</DocumentFragment>
`;
